File: solutions/ruby/run_solution.rb
require 'colorize'
require './minimum_distances'

#input parsing and reading
input00 = File.open("../../test-cases/input/input00.txt")
input01 = File.open("../../test-cases/input/input01.txt")

data = input00.map(&:chomp)

a00 = data[1].split(" ").map(&:to_i)

data = input01.map(&:chomp)

a01 = data[1].split(" ").map(&:to_i)

# problem application
start = Time.now
response00 = minimum_distances a00
response01 = minimum_distances a01
finish = Time.now

puts("solved all the test cases in " + (1000 * (finish - start)).round(2).to_s + "ms")

#expected outputs
start = Time.now
output00File = File.open("../../test-cases/output/output00.txt")
output01File = File.open("../../test-cases/output/output01.txt")

data = output00File.map(&:chomp)
output00 = data[0].to_i

data = output01File.map(&:chomp)
output01 = data[0].to_i

if response00 === output00
	puts "INPUT 00".green
else
	puts "INPUT 00".red
end

if response01 === output01
	puts "INPUT 01".green
else
	puts "INPUT 01".red
end

File: solutions/java/RunSolution.java
import java.io.*; 
import java.util.Arrays;
import java.math.BigInteger;

public class RunSolution{

	public static final String ANSI_RED = "\u001B[31m";
	public static final String ANSI_GREEN = "\u001B[32m";
	public static final String ANSI_RESET = "\u001B[0m";

	public static void main(String[] args) throws Exception{

		//input parsing and reading
		File input00File = new File("../../test-cases/input/input00.txt");
		BufferedReader brInput00 = new BufferedReader(new FileReader(input00File));

		File input01File = new File("../../test-cases/input/input01.txt");
		BufferedReader brInput01 = new BufferedReader(new FileReader(input01File));

		File input02File = new File("../../test-cases/input/input02.txt");
		BufferedReader brInput02 = new BufferedReader(new FileReader(input02File));

		String st;
		int currentLine = 0;
		String lineSplit[];

		int[] a00 = null;
		int[] a01 = null;
		int[] a02 = null;

		while((st = brInput00.readLine()) != null){
			if(currentLine == 1){
				a00 = Arrays
					.stream(st.split(" "))
					.mapToInt(Integer::parseInt)
					.toArray();
			}
			currentLine++;
		}

		currentLine = 0;

		while((st = brInput01.readLine()) != null){
			if(currentLine == 1){
				a01 = Arrays
					.stream(st.split(" "))
					.mapToInt(Integer::parseInt)
					.toArray();
			}
			currentLine++;
		}

		currentLine = 0;

		while((st = brInput01.readLine()) != null){
			if(currentLine == 1){
				a02 = Arrays
					.stream(st.split(" "))
					.mapToInt(Integer::parseInt)
					.toArray();
			}
			currentLine++;
		}

		// problem application

		int response00 = MinimumDistances.run(a00);
		int response01 = MinimumDistances.run(a01);
		int response02 = MinimumDistances.run(a02);
		

		// expected outputs

		File output00File = new File("../../test-cases/output/output00.txt");
		BufferedReader brOutput00 = new BufferedReader(new FileReader(output00File));

		File output01File = new File("../../test-cases/output/output01.txt");
		BufferedReader brOutput01 = new BufferedReader(new FileReader(output01File));

		File output02File = new File("../../test-cases/output/output02.txt");
		BufferedReader brOutput02 = new BufferedReader(new FileReader(output02File));

		int output00 = 0;
		int output01 = 0;
		int output02 = 0;

		while((st = brOutput00.readLine()) != null){
			output00 = Integer.parseInt(st.trim());
		}

		while((st = brOutput01.readLine()) != null){
			output01 = Integer.parseInt(st.trim());
		}

		while((st = brOutput02.readLine()) != null){
			output02 = Integer.parseInt(st.trim());
		}

		System.out.println(((response00 == output00) ? ANSI_GREEN : ANSI_RED) + "INPUT 00");
		System.out.println(((response01 == output01) ? ANSI_GREEN : ANSI_RED) + "INPUT 01");
		System.out.println(((response02 == output02) ? ANSI_GREEN : ANSI_RED) + "INPUT 02"+ ANSI_RESET);
	}
}